applicationset/generators/pull_request.go

```diff
@@ -69,10 +75,19 @@ func (g *PullRequestGenerator) GenerateParams(appSetGenerator *argoprojiov1alpha
 	}
 
 	pulls, err := pullrequest.ListPullRequests(ctx, svc, appSetGenerator.PullRequest.Filters)
+	params := make([]map[string]any, 0, len(pulls))
 	if err != nil {
-		return nil, fmt.Errorf("error listing repos: %w", err)
+		errStr := strings.ToLower(err.Error())
+		if (strings.Contains(errStr, "404") ||
+			strings.Contains(strings.ToLower(errStr), "not found")) && g.GetContinueOnRepoNotFoundError(appSetGenerator) {
```

Comment:
I am concerned that containing `not found` _or_ `404` can catch some unrelated scenarios. Mind the errors come from a number of providers, and those might not be http related errors.

Is there a better way to detect this?